cmd/podman/pods/stats.go

```diff
@@ -0,0 +1,191 @@
+package pods
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+	"text/template"
+	"time"
+
+	"github.com/buger/goterm"
+	"github.com/containers/buildah/pkg/formats"
+	"github.com/containers/libpod/cmd/podman/registry"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/containers/libpod/pkg/util/camelcase"
+	"github.com/spf13/cobra"
+)
+
+type podStatsOptionsWrapper struct {
+	entities.PodStatsOptions
+
+	// Format - pretty-print to JSON or a go template.
+	Format string
+	// NoReset - do not reset the screen when streaming.
+	NoReset bool
+	// NoStream - do not stream stats but write them once.
+	NoStream bool
+}
+
+var (
+	statsOptions     = podStatsOptionsWrapper{}
+	statsDescription = `For each specified pod this command will display percentage of CPU, memory, network I/O, block I/O and PIDs for containers in one the pods.`
+	// Command: podman pod _pod_
+	statsCmd = &cobra.Command{
+		Use:   "stats [flags] [POD...]",
+		Short: "Display a live stream of resource usage statistics for the containers in one or more pods",
+		Long:  statsDescription,
+		RunE:  stats,
+		Example: `podman pod stats
+  podman pod stats a69b23034235 named-pod
+  podman pod stats --latest
+  podman pod stats --all`,
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: statsCmd,
+		Parent:  podCmd,
+	})
+
+	flags := statsCmd.Flags()
+	flags.BoolVarP(&statsOptions.All, "all", "a", false, "Provide stats for all pods")
+	flags.StringVar(&statsOptions.Format, "format", "", "Pretty-print container statistics to JSON or using a Go template")
+	flags.BoolVarP(&statsOptions.Latest, "latest", "l", false, "Provide stats on the latest pod podman is aware of")
+	flags.BoolVar(&statsOptions.NoReset, "no-reset", false, "Disable resetting the screen when streaming")
+	flags.BoolVar(&statsOptions.NoStream, "no-stream", false, "Disable streaming stats and only pull the first result")
+
+	if registry.IsRemote() {
+		_ = flags.MarkHidden("latest")
+	}
+}
+
+func stats(cmd *cobra.Command, args []string) error {
+	// Validate input.
+	if err := entities.ValidatePodStatsOptions(args, &statsOptions.PodStatsOptions); err != nil {
+		return err
+	}
+
+	format := statsOptions.Format
+	doJson := strings.ToLower(format) == formats.JSONString
+	header := getPodStatsHeader(format)
+	stream := true
+
+	for stream {
```

Comment:
is `stream` modified anywhere?

Could it just be `for {`?

Reply:
Good catch!